lib/cisco_node_utils/cmd_ref/itd_device_group.yaml

```diff
@@ -0,0 +1,83 @@
+# itd_device_group
+---
+_exclude: [N3k, N5k, N6k, ios_xr]
+
+_template:
+  get_command: "show running-config all | section itd"
+  get_context:
+      - '/^itd device-group <name>/'
+      - '(?)/^node <ntype> <nname>/'
```

Comment:
If I understand this PR correctly, you're adding support for two types - itd_device_group and itd_device_group_node, and some of the below properties are specific to itd_device_group while others are specific to itd_device_group_node. If that's correct, rather than marking the 'node' as optional in the YAML (meaning itd_device_group properties may or may not have a node, and itd_device_group_node properties also may or may not have a node), I would strongly recommend that you create two separate YAML files - itd_device_group.yaml (which NEVER contains a 'node' line in the context) and itd_device_group_node.yaml (which ALWAYS contains a 'node' line in the context).